### ISSUE TITLE
[HOLD for payment 2024-01-17] [$500] Threads - When deleting the last unread message, the "New" marker appears for a second

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#31637]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.1.5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Open a chat
2. Send two messages
3. Click on "Mark as unread" on the first message
4. Delete the messages one by one

## Expected Result:
The new marker disappears with no delay

## Actual Result:
The new marker disappears after a delay

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/80457243-2963-4f49-af63-ecdf717d78d2

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e466a8c7c7ded031</li>
        <li>Upwork Job ID: 1726989610491375616</li>
        <li>Last Price Increase: 2023-12-04</li>
<li>Automatic offers: </li>
<ul>
<li>s-alves10 | Contributor | 27964548</li>
</ul></li>
    </ul>
</details>

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.


### What is the root cause of that problem?
We update the unread marker only after the action is actually deleted (the API response comes). Therefore, the marker is briefly visible while the API request is running.

### What changes do you think we should make in order to solve the problem?
We should hide the marker when the last action has has pending `DELETE` action.

For this, in this function, we should also return `false` if the `reportAction` has pending `DELETE` action:

```
if (reportAction.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
    return false;
}
```

https://github.com/Expensify/App/blob/8bb02a4715e0724499a9c8969b73ba34f0037d3e/src/pages/home/report/ReportActionsList.js#L334-L354

### What alternative solutions did you explore? (Optional)

**Comment by s-alves10** [comment]
This issue was addressed here https://github.com/Expensify/App/pull/30485#issuecomment-1784859939
I think the suggested solution would solve the issue

**Comment by MonilBhavsar** [comment]
Dupe of https://github.com/Expensify/App/issues/25606

**Comment by zanyrenney** [comment]
Closing as this is a dupe!

**Comment by MonilBhavsar** [comment]
@ArekChr any thoughts on the proposals from paultsimura and s-alves10 in another PR

**Comment by ArekChr** [comment]
As previously discussed, we agreed to address this issue in a separate PR. I've reviewed the solution provided by @s-alves10 in [PR #30485](https://github.com/Expensify/App/pull/30485#issuecomment-1784859939) and it solves the bug. Given that @s-alves10 has already developed a working fix, I think we should go ahead with this approach.

:ribbon: :eyes: :ribbon: C+ reviewed

**Comment by MonilBhavsar** [comment]
Thanks! Looks good, since we're ReportActionsUtils.ts, let's make sure we don't add any regression.

**Comment by s-alves10** [comment]
@ArekChr 

PR is ready for review https://github.com/Expensify/App/pull/32474

**Comment by MonilBhavsar** [comment]
This has caused regression. We're working on a fix

**Comment by MonilBhavsar** [comment]
PR is in review

**Comment by zanyrenney** [comment]
Not paying out yet due to regression.

**Comment by zanyrenney** [comment]
@MonilBhavsar can you let me know when the regression is resolved for payout please?

**Comment by MonilBhavsar** [comment]
PR is deployed to production yesterday. I'll add seven days and update the title

**Comment by zanyrenney** [comment]
not overdue yet, waiting for regression 7 day period.

**Comment by zanyrenney** [comment]
**Payment summary*

@ArekChr does not require payment (Contractor)
@s-alves10 requires payment [automatic offer](https://www.upwork.com/nx/wm/offer/27964548) (Contributor) - PAID $500.
